### ISSUE TITLE
Select emulator to launch with Archive Cache Manager

### ISSUE THREAD

**Issue by nixxou** [fraganator/archive-cache-manager#19]
When i right click, without using your plugin i have the option to choose the emulator i want to use.
![image](https://user-images.githubusercontent.com/45721836/161352879-b2c0b00e-39b9-4888-a7d8-30a0343bccb8.png)

I don't know if there is technical issue that prevent to implement it.

I will share the use case i'm dealing with :
For cartridges consoles, my roms are packed together with their clone & hacks into 7z and i tend to collect hacked version of populars games. So for Super Mario 64, i have most roms from this list : https://mario64hacks.fandom.com/wiki/List_of_Hacks
My issue is, most of them run only in Project 64 v1.6.
So if i could have an option like the filename priority who allow to write string with wildcard, per platform that will force an emulator instead of my default one, this would be great.

If that's technically doable, a dropdown list near the button Play & Cancel with the list of emulators for the platform (with the default one selected) could be nice too. You can make it appears only where there is more than one emulator registered for the platform.

**Comment by fraganator** [comment]
Hi @nixxou - I've added an emulator dropdown to the Select Rom In Archive window, which lists the emulators that support the platform of the selected game. The first entry in the dropdown will be the game's default emulator. You can try it out in the beta build over in [this comment](https://github.com/fraganator/archive-cache-manager/issues/13#issuecomment-1086665321).

There is the option in LaunchBox to set a default emulator on a per game basis. Would that cover your use case? Or are you looking to use a specific emulator for a game within the 7z archive? I don't know if the plugin can choose which emulator LaunchBox will use when the game is launched via the Play menu, but it can manually launch a game with a specific emulator like has just been implemented.

**Comment by nixxou** [comment]
@fraganator Hi. I was testing the feature, but i got an issue.
If i launch a game from a 7z using the beta, with a secondary optional emulator, i've got this command send : 
"C:\Program Files (x86)\Project64 1.6\Project64.exe" C:\coffre\7z64\Super Mario 64.7z
So it seems that he try to launch the whole archive as parameter instead of the rom itself.

Also, something like that could be nice, since there is no need to show the comboBox if there is no choice to make : 
```
  if(emulatorList.Count() <= 1)
  {
      emulatorComboBox.Visible = false;
      emulatorComboBoxLabel.Visible = false;
  }
```

**Comment by fraganator** [comment]
> If i launch a game from a 7z using the beta, with a secondary optional emulator, i've got this command send : "C:\Program Files (x86)\Project64 1.6\Project64.exe" C:\coffre\7z64\Super Mario 64.7z So it seems that he try to launch the whole archive as parameter instead of the rom itself.

Thanks, I think I know why this has happened. I'll fix it up asap.

**Comment by fraganator** [comment]
Hi @nixxou - This might be a silly question, but is _Extract ROMs in archive before running_ enabled for Project64 in the emulator settings?

**Comment by nixxou** [comment]
> Hi @nixxou - This might be a silly question, but is _Extract ROMs in archive before running_ enabled for Project64 in the emulator settings in LaunchBox?

My bad, this option was not enabled.

**Comment by fraganator** [comment]
Available in v2.12.
